test/external/external_benchmark_multitensor_allreduce.py

```diff
@@ -0,0 +1,56 @@
+import time
+import functools
+from tinygrad import Device
+from tinygrad.lazy import LazyBuffer
+from tinygrad.ops import ReduceOps, BinaryOps, LoadOps, GlobalCounters
+from tinygrad.features.multi import ring_allreduce
+from tinygrad.realize import create_schedule, run_schedule
+from tinygrad.dtype import dtypes
+from typing import List
+
+def naive_allreduce(op: ReduceOps, lbs: List[LazyBuffer]):
+  bop = {ReduceOps.SUM:BinaryOps.ADD, ReduceOps.MAX:BinaryOps.MAX}[op]
+  return [functools.reduce(lambda x,y: x.e(bop, y), [x.copy_to_device(lb.device) for x in lbs]) for lb in lbs]
+
+def realize(x: LazyBuffer | List[LazyBuffer]):
+    run_schedule(create_schedule(x if isinstance(x, list) else [x]))
+
+def test(impl, devs: List[str], N: int, iters:int = 10):
+  secs, gflops, gbs = 0, 0, 0
+  for i in range(-2, iters):
+    GlobalCounters.reset()
+    lbs = [LazyBuffer.loadop(LoadOps.CONST, (), dtypes.float, d, float(1+i)).reshape((1,)).expand((N,)).e(LoadOps.CONTIGUOUS) for i,d in enumerate(devs)] # noqa: E501
```

Comment:
can you use the shard API to construct the sharded tensor? I think it's more useful to test the construction using user facing api. Same for the unit test in test_multitensor

Reply:
It first creates it on some device and copies to all other devices afterwards which is slower + will eat more memory, with this it creates stuff directly on device where it should be.

Reply:
I can replace `LazyBuffer.loadop(LoadOps.CONST, (), dtypes.float, d, float(1+i)).reshape((1,)).expand((N,)).e(LoadOps.CONTIGUOUS)` with something like `Tensor.full((N,), float(1+i), device=d).contiguous()` though

Reply:
> Same for the unit test in test_multitensor

```python
t = Tensor.rand(tuple([(n if i == 0 else 1) * randint(1, 10) for i in range(randint(1, 4))])).shard_(tuple([d0, d1, d2, d3][:n]), 0)
```
I do it with shard here (because size is small and it doesn't matter here)